### PR TITLE
Issue #2157 pass type specific options for hostfolder add

### PR DESCRIPTION
**Pull request by anjannath**
-  implement `--options` flag for hostfolder add command

Need to test for CIFS and on Windows..

### CODE REVIEW

File: pkg/minishift/hostfolder/config/hostfolder_config.go [hunk 1]
```diff
@@ -17,12 +17,13 @@ limitations under the License.
...
-	UserName   = "username"
-	Password   = "password"
-	Domain     = "domain"
+	Source       = "source"
+	UncPath      = "uncpath"
+	MountPoint   = "mountpoint"
+	UserName     = "username"
+	Password     = "password"
+	Domain       = "domain"
+	ExtraOptions = "extraoptions"
```
Comment (gbraad) [inline pkg/minishift/hostfolder/config/hostfolder_config.go]: it looks better in JSON to use `extra-options`.

File: pkg/minishift/hostfolder/cifs_hostfolder.go [hunk 2]
```diff
@@ -63,11 +63,12 @@ func (h *CifsHostFolder) Mount(driver drivers.Driver) error {
...
 
 	cmd := fmt.Sprintf(
-		"sudo mount -t cifs %s %s -o username=%s,password=%s",
+		"sudo mount -t cifs %s %s -o username=%s,password=%s,%s",
 		h.config.Options[config.UncPath],
 		h.config.MountPoint(),
 		h.config.Options[config.UserName],
-		password)
+		password,
+		h.config.Option(config.ExtraOptions))
```
Comment (gbraad) [inline pkg/minishift/hostfolder/cifs_hostfolder.go]: why is this not `h.config.Options[config.ExtraOptions]` ?

File: pkg/minishift/hostfolder/sshfs_hostfolder.go [hunk 3]
```diff
@@ -75,11 +75,12 @@ func (h *SSHFSHostFolder) Mount(driver drivers.Driver) error {
 	// Mount command seems to fail occasionally. Give it a couple of attempts
 	mount := func() (err error) {
 		cmd := fmt.Sprintf(
-			"sudo sshfs docker@%s:%s %s -o IdentityFile=%s -o 'StrictHostKeyChecking=no' -o reconnect -o allow_other -o idmap=none -p %d",
+			"sudo sshfs docker@%s:%s %s -o IdentityFile=%s -o 'StrictHostKeyChecking=no' -o reconnect -o allow_other -o idmap=none %s -p %d",
 			ip,
 			h.config.Option(config.Source),
 			h.config.MountPoint(),
 			keyFile,
+			h.config.Option(config.ExtraOptions),
```
Comment (gbraad) [inline pkg/minishift/hostfolder/sshfs_hostfolder.go]: seems these two implementations aren't very consistent.

Comment (anjannath) [reply pkg/minishift/hostfolder/sshfs_hostfolder.go]: Yes, will change it to use the function in both cases.

Comment (anjannath) [reply pkg/minishift/hostfolder/sshfs_hostfolder.go]: @gbraad Done!

**Comment by amitkrout** [comment]
@anjannath  Would be great if you add some more scenarios based example

**Comment by LalatenduMohanty** [comment]
@anjannath You need to rebase this with master as I am getting some merge conflicts.

**Comment by gbraad** [comment]
Hasn't been updated

**Comment by anjannath** [comment]
@LalatenduMohanty @gbraad Sorry!! I missed your comments, re-basing now.

**Comment by amitkrout** [comment]
@anjannath would you please paste the expected output so that it would be easy to frame a test scenario if possible. If there is a possibility of test scenario then no need to wait for other pr to merge. IMO a followup integration test issue can be created for for later use and that would also be fine.

**Comment by anjannath** [comment]
@gbraad We need to add integration tests, as the functions cannot be unit tested, @amitkrout There won't be any specific o/p for the `hostfolder add` or `hostfolder mount` command without logs enabled.

You have to run the `m hostfolder mount` command with logging enabled and then, we can verify if the extra arguments were actually passed to the `sshfs` mount command.

**Comment by amitkrout** [comment]
>   -  implement `--options` flag for hostfolder add command

@anjannath Would you please add a example here i mean the execution result

> Need to test for CIFS and on Windows..

Have you tested it ? if yes remove it from description

**Comment by LalatenduMohanty** [comment]
@amitkrout  check https://github.com/minishift/minishift/issues/2157#issuecomment-408399335

**Pull request merged**